pkgs/by-name/dr/dragonflydb/package.nix

```
@@ -1,109 +1,252 @@
+    cp -r --no-preserve=mode,ownership ${jemalloc.src} build/deps-nixpkgs/jemalloc
+    cp -r --no-preserve=mode,ownership ${jsoncons-dragonfly} build/deps-nixpkgs/jsoncons
+    cp -r --no-preserve=mode,ownership ${liburing.src} build/deps-nixpkgs/uring
+    cp -r --no-preserve=mode,ownership ${lua-dragonfly} build/deps-nixpkgs/lua
+    cp -r --no-preserve=mode,ownership ${lz4.src} build/deps-nixpkgs/lz4
+    cp -r --no-preserve=mode,ownership ${mimalloc216} build/deps-nixpkgs/mimalloc216
+    cp -r --no-preserve=mode,ownership ${mimalloc224} build/deps-nixpkgs/mimalloc224
+    cp -r --no-preserve=mode,ownership ${pugixml.src} build/deps-nixpkgs/pugixml
+    cp -r --no-preserve=mode,ownership ${re-flex.src} build/deps-nixpkgs/reflex
+    cp -r --no-preserve=mode,ownership ${xxHash.src} build/deps-nixpkgs/xxhash
+    cp -r --no-preserve=mode,ownership ${zstd.src} build/deps-nixpkgs/zstd
+
+    # c-ares is provided as a tarball, extract it
+    mkdir -p build/deps-nixpkgs/cares
+    tar -xzf ${c-ares.src} -C build/deps-nixpkgs/cares --strip-components=1
+
+    ${
+      if withAws then
+        ''
+          cp -r --no-preserve=mode,ownership ${aws-sdk-cpp-1-11-162} build/deps-nixpkgs/aws-sdk-cpp
+        ''
+      else
+        ""
+    }
+
+    ${
+      if withGcp then
+        ''
+          cp -r --no-preserve=mode,ownership ${rapidjson.src} build/deps-nixpkgs/rapidjson
+        ''
+      else
+        ""
+    }
+
+    ${
+      if withGperf then
+        ''
+          cp -r --no-preserve=mode,ownership ${gperftools.src} build/deps-nixpkgs/gperf
+        ''
+      else
+        ""
+    }
+
+    ${
+      if withSearch then
+        ''
+          cp -r --no-preserve=mode,ownership ${hnswlib.src} build/deps-nixpkgs/hnswlib
+          cp -r --no-preserve=mode,ownership ${uni-algo.src} build/deps-nixpkgs/uni-algo
+        ''
+      else
+        ""
+    }
+
+    # Fix permissions
+    chmod -R u+w build/deps-nixpkgs build/_deps
+    chmod u+x build/deps-nixpkgs/reflex/configure
+    chmod u+x build/deps-nixpkgs/uring/configure
+    touch build/deps-nixpkgs/xxhash/xxhsum.1
   '';
 
   nativeBuildInputs = [
     autoconf
     autoconf-archive
-    automake
+    automake116x
+    bison
     cmake
     ninja
   ];
 
   buildInputs = [
     boost
-    libunwind
     libtool
     openssl
-  ];
+    zlib
+  ]
+  ++ (if withPcre then [ pcre2 ] else [ ])
+  ++ (if withRe2 then [ re2 ] else [ ])
+  ++ (if withUnwind then [ libunwind ] else [ ]);
 
   cmakeFlags = [
     "-DCMAKE_AR=${gcc-unwrapped}/bin/gcc-ar"
     "-DCMAKE_RANLIB=${gcc-unwrapped}/bin/gcc-ranlib"
+    "-DENABLE_GIT_VERSION=OFF"
+    "-DWITH_ASAN=${if withAsan then "ON" else "OFF"}"
+    "-DWITH_AWS=${if withAws then "ON" else "OFF"}"
+    "-DWITH_GCP=${if withGcp then "ON" else "OFF"}"
+    "-DWITH_GPERF=${if withGperf then "ON" else "OFF"}"
+    "-DWITH_SEARCH=${if withSearch then "ON" else "OFF"}"
+    "-DWITH_USAN=${if withUsan then "ON" else "OFF"}"
   ];
```

One nit I didn't notice earlier, take it or leave it.

```suggestion
  cmakeFlags = [
    (lib.cmakeFeature "CMAKE_AR" "${gcc-unwrapped}/bin/gcc-ar")
    (lib.cmakeFeature "CMAKE_RANLIB" "${gcc-unwrapped}/bin/gcc-ranlib")
    (lib.cmakeBool "ENABLE_GIT_VERSION" false)
    (lib.cmakeBool "WITH_ASAN" withAsan)
    (lib.cmakeBool "WITH_AWS" withAws)
    (lib.cmakeBool "WITH_GCP" withGcp)
    (lib.cmakeBool "WITH_GPERF" withGperf)
    (lib.cmakeBool "WITH_SEARCH" withSearch)
    (lib.cmakeBool "WITH_USAN" withUsan)
  ];
```